x/gov/keeper/proposal.go

```diff
@@ -191,3 +193,19 @@ func (keeper Keeper) ActivateVotingPeriod(ctx sdk.Context, proposal types.Propos
 	keeper.RemoveFromInactiveProposalQueue(ctx, proposal.ProposalID, proposal.DepositEndTime)
 	keeper.InsertActiveProposalQueue(ctx, proposal.ProposalID, proposal.VotingEndTime)
 }
+
+func (keeper Keeper) MarshalProposal(proposal types.Proposal) []byte {
```

Comment:
Ideally we have `Must` variants of these methods in case there are places where we want to gracefully handle the error.

Reply:
Aren't we trying to get away from panicking on errors?

Reply:
Absolutely not. Any decoding or encoding error in the keeper should panic -- the state-machine cannot proceed in this case.

However, since we now use the keeper as also a psuedo-codec, then we need to have non-panic variants so things like queriers and other client logic can use them.

Reply:
Okay, adding `Must` variants